src/main/java/ezschedule/storage/JsonSerializableScheduler.java

```diff
@@ -52,7 +52,12 @@ public Scheduler toModelType() throws IllegalValueException {
             if (scheduler.hasEvent(event)) {
                 throw new IllegalValueException(MESSAGE_DUPLICATE_EVENT);
             }
-            scheduler.addEvent(event);
+            if (event.isPastEvent()) {
+                event.setCompleted();
+                scheduler.addEvent(event);
+            } else {
+                scheduler.addEvent(event);
+            }
```

Comment:
You can DRY `scheduler.addEvent(event)` so that it looks like
```java
if (event.isPastEvent()) {
    event.setCompleted();
}
 scheduler.addEvent(event);
```

Reply:
I have resolved the need by implementing `event.getCompletedStatus()` only in EventCard class